### DISCUSSION TITLE
What is usage difference between: job.status == 'success' and if :success() ?

### DISCUSSION THREAD

**Discussion by jacek-jablonski** [community/community#25173] [Actions]
<p>Hi,</p>
<p>what is possible usage difference between:</p>
<pre><code>if: job.status == 'success'
</code></pre>
<p>and</p>
<pre><code>if: success()
</code></pre>
<p>Thanks.</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p><strong>job.status</strong> is a context of the <strong>current</strong> job. Possible values are success, failure, or cancelled. <strong>job.status</strong> could be used not only in if conditional but also in other places, such as using it as an input variable value of an action.  </p>
<pre><code>- name: Dump job context
        env:
          JOB_STATUS: ${{ job.status }}
        run: echo "$JOB_STATUS"
</code></pre>
<p><strong>success()</strong> is a check function which could be used as an expression in if conditionals.</p>
<p>When they are used in <strong>jobs.&lt;job_id&gt;.steps.if</strong>  conditional, there is no difference between them. You could choose any one you like.  </p>
<p><strong>success()</strong> could be used in jobs.&lt;job_id&gt;.if , but <strong>job.status</strong>  could not.</p>

**Comment by jacek-jablonski** [comment]
<p>Many thanks for explanation.</p>
